src/web/query_socket.cpp

```diff
@@ -52,6 +52,15 @@ query_socket::query_socket(zmq::context& context, server_node& node,
         request.enqueue(data_chunk{});
     };
 
+    const auto encode_height = [](zmq::message& request,
+        const std::string& command, const std::string& arguments, uint32_t id)
+    {
+        request.enqueue(command);
+        request.enqueue_little_endian(id);
+        const uint32_t height = std::strtoul(arguments.c_str(), nullptr, 0);
```

Comment:
I don't understand the multi-argument encoding within the string here. But `std::strtoul` should be avoided here. Passing the c string directly is Unicode unsafe.

Reply:
There is an implicit cast here from `unsigned long` to `uint32_t`. These aren't the same size, making this a bug unless the argument is first validated.

Reply:
It doesn't appear that there's any way using these handlers to return an error for invalid parameterization.

Reply:
> It doesn't appear that there's any way using these handlers to return an error for invalid parameterization.

I'm adding that, there's no reason there shouldn't be error handing on failed encodings.